### PR TITLE
Fixing "C4127: conditional expression is constant" Visual Studio 2022 warning in pedantic mode

### PR DESCRIPTION
**Pull request by chronoxor**

### CODE REVIEW

File: include/fmt/format.h [hunk 1]
```diff
@@ -2260,7 +2260,7 @@ FMT_CONSTEXPR20 auto write(OutputIt out, T value,
     precision = 1;
   }
   if (const_check(std::is_same<T, float>())) fspecs.binary32 = true;
-  if (!std::numeric_limits<T>::is_iec559 || std::numeric_limits<T>::digits > 64)
+  if (const_check(!std::numeric_limits<T>::is_iec559) || std::numeric_limits<T>::digits > 64)
```
Comment (phprus) [inline include/fmt/format.h:2263]: Maybe:
```c++
if (const_check(!std::numeric_limits<T>::is_iec559 || std::numeric_limits<T>::digits > 64))
```
?

``std::numeric_limits<T>::digits`` is constexpr too.

Comment (vitaut) [reply include/fmt/format.h:2263]: and please apply clang-format

Comment (chronoxor) [reply include/fmt/format.h]: Fixed

**Comment by vitaut** [comment]
Thanks

**Pull request merged**